frontend/src/uxbox/main/ui/workspace/rules.cljs

```diff
@@ -2,156 +2,129 @@
-
-      (< (mod value mid-ticks-mod) step-size)
-      (conj acc (str/format "M %s %s L %s %s" pos 10 pos step-padding))
-
-      :else
-      (conj acc (str/format "M %s %s L %s %s" pos 15 pos step-padding)))))
-
-(defn- make-horizontal-tick
-  [zoom acc value]
-  (let [big-ticks-mod (big-ticks-mod zoom)
-        mid-ticks-mod (mid-ticks-mod zoom)
-        pos (+ (* value zoom)
-               scroll-padding)]
-    (cond
-      (< (mod value big-ticks-mod) step-size)
-      (conj acc (str/format "M %s %s L %s %s" 5 pos step-padding pos))
-
-      (< (mod value mid-ticks-mod) step-size)
-      (conj acc (str/format "M %s %s L %s %s" 10 pos step-padding pos))
-
-      :else
-      (conj acc (str/format "M %s %s L %s %s" 15 pos step-padding pos)))))
-
-;; --- Horizontal Text Label
-
-(mf/defc horizontal-text-label
-  [{:keys [zoom value] :as props}]
-  (let [big-ticks-mod (big-ticks-mod zoom)
-        pos (+ (* value zoom)
-               rule-padding
-               scroll-padding)]
-    (when (< (mod value big-ticks-mod) step-size)
-      [:text {:x (+ pos 2)
-              :y 13
-              :key (str pos)
-              :fill "#9da2a6"
-              :style {:font-size "12px"}}
-       value])))
-
-;; --- Horizontal Text Label
-
-(mf/defc vertical-text-label
-  [{:keys [zoom value] :as props}]
-  (let [big-ticks-mod (big-ticks-mod zoom)
-        pos (+ (* value zoom)
-               scroll-padding)]
-    (when (< (mod value big-ticks-mod) step-size)
-      [:text {:y (- pos 3)
-              :x 5
-              :key (str pos)
-              :fill "#9da2a6"
-              :transform (str/format "rotate(90 0 %s)" pos)
-              :style {:font-size "12px"}}
-       value])))
-
-;; --- Horizontal Rule Ticks (Component)
-
-(mf/defc horizontal-rule-ticks
-  {:wrap [mf/memo]}
-  [{:keys [zoom]}]
-  (let [path (reduce (partial make-vertical-tick zoom) [] +ticks+)]
-    [:g
-     [:path {:d (str/join " " path)}]
-     (for [tick +ticks+]
-       [:& horizontal-text-label {:zoom zoom :value tick :key tick}])]))
-
-;; --- Vertical Rule Ticks (Component)
-
-(mf/defc vertical-rule-ticks
-  {:wrap [mf/memo]}
-  [{:keys [zoom]}]
-  (let [path (reduce (partial make-horizontal-tick zoom) [] +ticks+)]
-    [:g
-     [:path {:d (str/join " " path)}]
-     (for [tick +ticks+]
-       [:& vertical-text-label {:zoom zoom :value tick :key tick}])]))
-
-;; --- Horizontal Rule (Component)
+   [uxbox.util.object :as obj]))
+
+(def STEP-PADDING 20)
 
 (mf/defc horizontal-rule
-  {:wrap [mf/memo]}
-  [props]
-  (let [scroll (use-rxsub  ms/viewport-scroll)
-        zoom (mf/deref refs/selected-zoom)
-        translate-x (- (- scroll-padding) (:x scroll))]
-    [:svg.horizontal-rule
-     {:width c/viewport-width
-      :height 20}
-     [:rect {:height 20
-             :width c/viewport-width}]
-     [:g {:transform (str "translate(" translate-x ", 0)")}
-      [:& horizontal-rule-ticks {:zoom zoom}]]]))
+  [{:keys [zoom size]}]
```

Comment:
Both horizontal and vertical rules are almos identical. I think we should try to "merge" both or at leas get some common functions so we don't have to maintain through copy/paste :smile:

Reply:
We'll refactor this part in a following MR